### ISSUE TITLE
Make a release

### ISSUE THREAD

**Issue by BenMcLean** [droidmjt/Droidsound#544]
Can you please make releases on Github?

**Comment by BenMcLean** [comment]
And also check in your code ... just generally?

**Comment by droidmjt** [comment]
no and no, sorry

**Comment by BenMcLean** [comment]
OK, but isn't some of the code here based on previously existing code with license requirements that require derivatives to be open source?

**Comment by nathanmp** [comment]
Agreed. Not releasing the source code violates the licenses of both ffmpeg and vgmplay. And locking GitHub issues doesn't change that.

**Comment by drew887** [comment]
Yeah it looks like in VGMPlay's case at least they have their own licence issues going on with several of their dependencies using different licences, but it does seem like the consensus is that it should be some form of GPL licence. 

But yeah FFMPEG for sure needs to be updated, the header files here contain the licence text at the top of them, but you're not distributing a copy of the GPL 2.1 and seem to have instead stripped it out. 

Since FFMPEG wasn't included [the repository of the original project](https://github.com/sasq64/droidsound) I do believe you're the one legally liable here to include and distribute the GPL 2.1 licence text along with this repo. 

I could keep checking the other ones but @nathanmp's find of the FFMPEG one is enough to cause concern without looking further

**Comment by nathanmp** [comment]
@drew887 Yeah, understanding is that GPL doesn't require that every change is put in version control immediately. But I'm pretty sure it requires that the current source code for each release is distributed.

I also want to add this - it's a bit unnerving that the source code hasn't been released for years, that the apks are only hosted on Dropbox, and that the app uses fairly intrusive permissions. I understand there are technical reasons for the permissions - but how does anyone know those are the ONLY reasons? More bluntly, the project could be legitimately using them to play music while also taking user data for some purpose. I decompiled the apk to smali files and did a quick grep for terms like "btc" and "Bitcoin" to see if there was any mining going on but I didn't see any. I still need to check if there are any hardcoded IP addresses, though.

**Comment by nathanmp** [comment]
> I mean I have faith in @droidmjt that nothing evil is going on here
I do too, and as far as I know the rationale for those permissions is sound. But when my issue was closed without comment, despite licensing requirements, I became a bit concerned, which is why I decompiled it. I was partly saying how I think some people would interpret the situation, given the current way it's distributed.
>and yeah the GPL does require that the source code is available upon request but I don't believe it has to be made available here every release.
You're right it doesn't have to made on every release, my bad. It has to be available on request (it seemingly isn't) and noted as available on request (it definitely isn't).
> My point was more that in the code that is currently live here, we have versions of these libraries that require their respective licenses to be distributed with them, so even if droid doesn't want to update the source code thats live I do believe they are required to add the licences those libraries expect to this repo.
To comply with GPL3:
1. The licenses of all components must be distributed.
2. The source code of the whole work - that is, droidsound, must be marked as available on request or distributed without request.
Maybe that's what you were saying with the last paragraph, not entirely sure. 
>even if droid doesn't want to update the source code 
I suppose I only asked him to make an update, which isn't the technical requirement. But he must provide it on request at a minimum. So I should - and am - explicitly requesting that he release an archive of the code of the most recent release as of now. He might be worried about the quality of the code, but I don't mind about that. But since he's using GPL3 libraries, whether he wants to do that or not is meaningless.
